packages/core/src/tools/error.spec.ts

```diff
@@ -44,32 +63,118 @@ describe('formatUnknownError', () => {
   })
 
   it('should format an error with an empty message', () => {
-    const stack: StackTrace = {
+    const stackTrace: StackTrace = {
       message: '',
       name: 'TypeError',
       stack: [],
     }
 
-    const formatted = formatUnknownError(stack, undefined, 'Uncaught')
+    const formatted = computeRawError({
+      ...DEFAULT_RAW_ERROR_PARMS,
+      stackTrace,
+      originalError: undefined,
+      handling: ErrorHandling.HANDLED,
+    })
 
     expect(formatted.message).toEqual('Empty message')
   })
 
   it('should format a string error', () => {
-    const errorObject = 'oh snap!'
+    const error = 'oh snap!'
 
-    const formatted = formatUnknownError(NOT_COMPUTED_STACK_TRACE, errorObject, 'Uncaught')
+    const formatted = computeRawError({
+      ...DEFAULT_RAW_ERROR_PARMS,
+      stackTrace: NOT_COMPUTED_STACK_TRACE,
+      originalError: error,
+      handling: ErrorHandling.HANDLED,
+    })
 
     expect(formatted.message).toEqual('Uncaught "oh snap!"')
   })
 
+  it('should compute stack trace form nested error, and not use param stacktrace', () => {
+    const stackTrace: StackTrace = {
+      message: 'some typeError message',
+      name: 'TypeError',
+      stack: [],
+    }
+
+    const error = new Error('foo') as ErrorWithCause
+    const nestedError = new Error('bar')
+
+    error.cause = nestedError
+
+    const formatted = computeRawError({
+      ...DEFAULT_RAW_ERROR_PARMS,
+      stackTrace,
+      originalError: error,
+      handling: ErrorHandling.HANDLED,
+    })
+
+    expect(formatted?.type).toEqual('TypeError')
+    expect(formatted?.message).toEqual('some typeError message')
+    expect(formatted?.causes?.[0].type).toEqual('Error')
+    expect(formatted?.causes?.[0].message).toEqual('bar')
+  })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏instead of having a dedicated test for it, what about enriching the existing `should compute an object error with causes`?
We could override error stack to validate that each stack is at the right place
```ts
    const error = new Error('foo: bar') as ErrorWithCause
    error.stack = 'Error: foo: bar\n    at <anonymous>:1:15'
   
    const nestedError = new Error('biz: buz') as ErrorWithCause
    nestedError.stack = 'NestedError: biz: buz\n    at <anonymous>:1:15'

...

    expect(formatted.type).toBe('Error')
    expect(formatted.stack).toBe(...)
    expect(causes[0].type).toBe('NestedError')
    expect(causes[0].stack).toBe(...)
```